Adding shortcode for accordion component (Bootstrap 5) (#1391)
google/docsy: userguide/content/en/docs/adding-content/shortcodes/index.md

If you are using this shortcode, note that when evaluating the conditions, substring matches are matches as well. That means, if you set `include-if="foobar"`, and `buildcondition = "foo"`, you have a match!

## Accordion

> Sometimes it's very useful to have an accordion element at hand when authoring content. One common use-case is FAQ section where only one section (= accordion item) will be open at any time. The following sample showcases a accordion presenting the four principles of object-oriented programming: 

Reviewer: ```suggestion
Sometimes it's very useful to have an accordion element at hand when authoring content. One common use case is a FAQ section where only one section (= accordion item) will be open at any time. The following sample showcases an accordion presenting the four principles of object-oriented programming: 
```

Author: Changed as requested.

> ### Sample accordion: Object-oriented programming

Reviewer: ```suggestion
### Sample accordion: object-oriented programming
```
changed `Object-oriented` to `object-oriented` to match case in previous section

Author: Changed as requested.

{{% /accordion-item %}}
{{% accordion-item header="Principle 3:&nbsp;**Inheritance**" %}}
**Inheritance** is the ability of creating a new class from an existing class. A child object acquires the property of a parent object. By means of inheritance, we achieve that a child class (subclass) acquires the properties and behavior of another parent class (super-class).
{{% /accordion-item %}}
{{% accordion-item header="Principle 4:&nbsp;**Polymorphism**" %}}
**Polymorphism** is derived from two Greek words: _πολύς_ (`polús`) translates to `many` and _μορφή_ (`morphḗ`) to `form`. So polymorphism means `many forms`. A child/subclass can define its own unique behavior and still share the same functionalities or behavior of its parent/base class. A subclass can have their own behavior and share some of its behavior from its parent class not the other way around. A parent class cannot have the behavior of its subclass.
{{% /accordion-item %}}
{{% /accordion %}}

> The Docsy template provides two shortcodes `accordion` and `accordion-item` that let you easily create an accordion component. To see how to use them, have a look at the following code block, which renders to the accordion shown above:

Reviewer: ```suggestion
The Docsy template provides two shortcodes, `accordion` and `accordion-item`, that let you easily create an accordion component. To see how to use them, have a look at the following code block, which renders to the accordion shown above:
```

Author: Added commas as requested.

```go-html-template
{{%/* accordion alwaysOpen=false */%}}
{{%/* accordion-item header="Principle 1: Abstraction" */%}}
**Abstraction** is the process of ...
{{%/* /accordion-item */%}}
{{%/* accordion-item header="Principle 2: Encapsulation" */%}}
**Encapsulation** means wrapping up data ...
{{%/* /accordion-item */%}}
{{%/* accordion-item Principle 3: Inheritance */%}}
**Inheritance** is the ability of creating a new class ...
{{%/* /accordion-item */%}}
... fourth item omitted
{{%/* /accordion %}}
```

### Shortcode details

An accordion component is implemented using two shortcodes:

> * The `accordion` shortcode, which is the container element for the collapsible items of the accordion. This shortcode can hold the optional named parameters `alwaysOpen`, and `flush`. Specify `alwaysOpen=true` if you want to make accordion items stay open even when another item is opened.

Reviewer: ```suggestion
* The `accordion` shortcode, which is the container element for the collapsible items of the accordion. This shortcode can hold the optional named parameters `alwaysOpen` and `flush`. Specify `alwaysOpen=true` if you want to make accordion items stay open even when another item is opened.
```

Author: Removed the superfluous comma, thanks for spotting this!

> * The various `accordion-item` shortcodes represent the collapsible items of your accordion. Specify the named parameter `header` for each `accordion-item` in order to set the header text of the item. If the `header` parameter is the only parameter inside your `accordion-item` shortcode, you can specify the header as unnamed parameter, something like `{{</* accordion-item "My header" */>}} … {{</* /accordion-item */>}}`. If your `accordion-item` shortcode does not have any parameters, the header of the item will default to `Header n`. When specifying `open=true` for an `accordion-item`, this item will be initially rendered opened (as opposed to an default collapsed item). If your item content is markdown, use the percent sign `%` as outermost delimiter of your `accordion-item` shortcode, your markup should look like `{{%/* accordion-item */%}}`Your \*\*markdown\*\* content`{{%/* /accordion-item */%}}`. In case of HTML content, use square brackets `<>` as outermost delimiters: `{{</* accordion-item */>}}`Your &lt;b&gt;HTML&lt;/b&gt; content`{{</* /accordion-item */>}}`.

Reviewer: ```suggestion
* The various `accordion-item` shortcodes represent the collapsible items of your accordion. Specify the named parameter `header` for each `accordion-item` in order to set the header text of the item. If the `header` parameter is the only parameter inside your `accordion-item` shortcode, you can specify the header as unnamed parameter, something like `{{</* accordion-item "My header" */>}} … {{</* /accordion-item */>}}`. If your `accordion-item` shortcode does not have any parameters, the header of the item will default to `Header n`. When specifying `open=true` for an `accordion-item`, this item will be initially rendered opened (as opposed to a default collapsed item). If your item content is markdown, use the percent sign `%` as the outermost delimiter of your `accordion-item` shortcode. Your markup should look like `{{%/* accordion-item */%}}`Your \*\*markdown\*\* content`{{%/* /accordion-item */%}}`. In case of HTML content, use angle brackets `<>` as the outermost delimiters: `{{</* accordion-item */>}}`Your &lt;b&gt;HTML&lt;/b&gt; content`{{</* /accordion-item */>}}`.
```
- minor punctuation updates
- changed "square brackets" to "angle brackets"

Author: Changed as requested.
